### PR TITLE
Honor productionOnly for browser routing

### PR DESCRIPTION
**Pull request by davidthor**
productionOnly option wasn't being respected client-side making it difficult to test

### CODE REVIEW

File: src/gatsby-browser.js [hunk 1]
```diff
@@ -1,5 +1,15 @@
-export function onRouteUpdate({ location }) {
-    if (process.env.NODE_ENV !== 'production' || !Array.isArray(window._hsq)) {
+import defaultOptions from './default-options';
+
+export function onRouteUpdate({ location }, pluginOptions) {
+    const { productionOnly } = Object.assign(
+        defaultOptions,
+        pluginOptions
+    );
```
Comment (AustinLeeGordon) [inline src/gatsby-browser.js]: This needs to be changed to the following:

```jsx
    const { productionOnly } = Object.assign(
		{},
        defaultOptions,
        pluginOptions
    );
```

Adding an extra empty object as the first parameter to `Object.assign` prevents `defaultOptions` from being mutated. We also need to make this change in `gatsby-ssr.js` if you'd like to update it in the same PR!

**Review by AustinLeeGordon — CHANGES_REQUESTED** [review]
@davidthor Looks great! I added a small change to prevent `defaultOptions` from being mutated. Let me know if you're able to make that update!

**Comment by davidthor** [comment]
> @davidthor Looks great! I added a small change to prevent `defaultOptions` from being mutated. Let me know if you're able to make that update!

Should be good to go now!

**Review by AustinLeeGordon — APPROVED** [review]

**Comment by AustinLeeGordon** [comment]
@davidthor Perfect, thanks! Had to make the same `Object.assign` fix in `gatsby-ssr.js` to fix the tests. Thanks for catching that issue and contributing! I'll get this merged and released.

**Comment by davidthor** [comment]
Thanks for the prompt reply and integration!

**Pull request merged**